home-manager/home-manager

```diff
@@ -87,14 +86,16 @@ function doBuild() {
 }
 
 function doSwitch() {
-    local wrkdir
-    wrkdir="$(mktemp -d)"
+    local wrkdir="$(mktemp -d)"
 
-    if doBuild "$wrkdir/generation" ; then
-        "$wrkdir/generation/activate"
-    fi
+    local generation
+    local exitCode=0
+    generation=$(doBuild "$wrkdir/result") && $generation/activate || exitCode=1
```

Comment:
With your verbose snippet, the script will actually succeed when `activate` fails.
Errors in functions don't cause `set -e` termination, only top-level commands.
The following monstrosity should work:

```bash
if generation=$(doBuild "$wrkdir/result"); then
    if ! $generation/activate; then
        exitCode=1
    fi
else
    exitCode=1
fi
```

Besides that: `generation` is always a store path, so I'd leave the quotes out.